File: AppDev_Act9/app/src/main/java/com/example/appdev_act9/MainActivity.java
package com.example.appdev_act9;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private Button btnAddition;
    private Button btnSubtraction;
    private Button btnMultiplication;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnAddition = findViewById(R.id.btn1);
        btnSubtraction = findViewById(R.id.btn2);
        btnMultiplication = findViewById(R.id.btn3);

        btnAddition.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startGameActivity("addition");
            }
        });

        btnSubtraction.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startGameActivity("subtraction");
            }
        });

        btnMultiplication.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startGameActivity("multiplication");
            }
        });
    }

    private void startGameActivity(String operation) {
        Intent intentGame = new Intent(this, game.class);
        intentGame.putExtra("operation", operation);
        startActivity(intentGame);
        finish();
    }
}

File: AppDev_Act7/app/src/main/java/com/example/appdev_act7_bmicalculator/MainActivity.kt
package com.example.appdev_act7_bmicalculator

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.appdev_act7_bmicalculator.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private var binding: ActivityMainBinding? = null
    private var choiceGender: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)

        setContentView(binding!!.root)

        binding!!.maleBtn.setOnClickListener { view ->
            if (binding!!.maleBtn.isChecked) {
                choiceGender = "MR."
                binding!!.FemaleBtn.isChecked = false
            } else {
                binding!!.maleBtn.isChecked = true
            }
        }

        binding!!.FemaleBtn.setOnClickListener { view ->
            if (binding!!.FemaleBtn.isChecked) {
                choiceGender = "MRS."
                binding!!.maleBtn.isChecked = false
            } else {
                binding!!.FemaleBtn.isChecked = true
            }
        }


        binding!!.subBtn.setOnClickListener { view ->
            val firstName = binding!!.editFirst.text.toString()
            val middleInitial = binding!!.editMI.text.toString()
            val lastName = binding!!.editLast.text.toString()
            val heightStr = binding!!.editCM.text.toString()
            val weightStr = binding!!.editKG.text.toString()

            if (!firstName.isEmpty() && !lastName.isEmpty() && !middleInitial.isEmpty() &&
                !heightStr.isEmpty() && !weightStr.isEmpty()
            ) {
                val height = heightStr.toFloat()
                val weight = weightStr.toFloat()

                if (choiceGender == null) {
                    Toast.makeText(
                        applicationContext,
                        "SELECT GENDER, THANK YOU!",
                        Toast.LENGTH_SHORT
                    ).show()
                    return@setOnClickListener
                }

                val compbmi = computeBMI(height, weight)

                val intent: Intent = if (compbmi < 18.5) {
                    Intent(this, UnderweightActivity::class.java)
                } else if (compbmi < 25) {
                    Intent(this, HealthyActivity::class.java)
                } else if (compbmi < 30) {
                    Intent(this, OverweightActivity::class.java)
                } else {
                    Intent(this, ObeseActivity::class.java)
                }


                intent.putExtra("FIRSTNAME", firstName)
                intent.putExtra("MIDDLE INITIAL", middleInitial)
                intent.putExtra("LASTNAME", lastName)
                intent.putExtra("GENDER", choiceGender)
                intent.putExtra("BMI", compbmi)
                startActivity(intent)
                finish()

            } else {
                Toast.makeText(
                    applicationContext,
                    "PLEASE INSERT CORRECTLY!",
                    Toast.LENGTH_SHORT
                ).show()
            }
        }
    }


    private fun computeBMI(height: Float, weight: Float): Float {
        val heightInMeters = height / 100
        return if (heightInMeters > 0) weight / (heightInMeters * heightInMeters) else 0.0f
    }
}